vignettes/wrapping-apis.Rmd

```diff
@@ -213,92 +224,130 @@ faker_person <- function(gender = NULL, birthday_start = NULL, birthday_end = NU
 
-Next we'll take a look at the NYTimes [Books API](https://developer.nytimes.com/docs/books-product/1/overview). It requires a very simple authentication with an API key that's included in every request.
-When you're wrapping an API that has a key you're going to face two struggles:
+This system is probably overkill if you only have one secret that you need to share in one or two places.
+But you almost invariably accumulate more secrets over time, and more people and computers that you need to share it with, so I think spending little time to understand this system and set up it for your package will pay off in the long term.
 
--   How do you test your package without sharing your key with the whole world?
+### Basics
 
--   How do you allow your users to supply their own key, without having to pass it to every function?
+httr2 provides `secret_encrypt()` and `secret_decrypt()` to scramble secrets so that you can include them in your public source code without worrying that others can read them.
+There are three basic steps to this process:
 
-I'll start by tackling the first problem because otherwise there's no way for me to show how the API works in this vignette 😃.
-We'll come back to the second at the very end of this section, because it's easiest to tackle once we have a function in place.
+1.  You create an **encryption** key with `secret_make_key()` that is used to scramble and descramble secrets using symmetric cryptography:
 
-### Encrypting secrets
+    ```{r}
+    key <- secret_make_key()
+    key
+    ```
 
-httr2 provides the `secret_decrypt()` and `secret_decrpyt()` functions that use asymmetric cryptography.
-To use them you need to first create a "key" that's used to lock and unlock the secrets.
-The best key is random because there's no way for anyone to guess it.
-httr2 provides `secret_make_key()` to make a secure key for you:
+    (Note that `secret_make_key()` uses a cryptographically secure random number generator provided by OpenSSL; it is not affected by R's RNG settings, and there's no way to make it reproducible.)
 
-```{r}
-secret_make_key()
-```
+2.  You scramble your secrets with `secret_encrypt()` and store the resulting text directly in the source code of your package:
 
-(Don't copy and paste this key; run `secret_make_key()` yourself to make your own personal key.)
+    ```{r}
+    secret <- secret_encrypt("this is a secret message", key)
+    secret
+    ```
 
-Now you need to make this key available in the places were you want to unlock the secrets.
-httr2 provides tools to retrieve keys from environment variables because this is a common way of communicating these sort of secrets.
-So start by opening your `.Renviron` (e.g. with `usethis::edit_r_environ()`) and add a line like:
+3.  When needed, you descramble the the secret using `secret_decrypt()`:
+
+    ```{r}
+    secret_decrypt(secret, key)
```

Comment:
```suggestion
    secret_decrypt(secret_scrambled, key)
```

Reply:
It feels like a concrete example is needed around here, but maybe that gets taken care of soon enough. Specifically, the code above makes it look like you'll store the scrambled version in a variable, but IRL these will be string literals.